tracetypes/org.eclipse.tracecompass.incubator.traceevent.core/src/org/eclipse/tracecompass/incubator/internal/traceevent/core/event/TraceEventAspects.java

```diff
@@ -159,7 +159,14 @@ public boolean isHiddenByDefault() {
         @Override
         public @Nullable Integer resolve(@NonNull ITmfEvent event) {
             if (event instanceof TraceEventEvent) {
-                return ((TraceEventEvent) event).getField().getTid();
+                Object tid = ((TraceEventEvent) event).getField().getTid();
+                if (tid instanceof Integer) {
+                    return (Integer) tid;
+                }
+                if (tid != null) {
```

Comment:
Not knowing the traces, in this case is tid a name or a number string? if it is a number string you could do parseInt() or something like that.

Add a comment why it's done like that.

Reply:
I think the new way is better.